containers/Dockerfile

```diff
@@ -0,0 +1,47 @@
+ARG PYTHON_VERSION=3
+FROM docker.io/python:${PYTHON_VERSION}-slim AS base
+
+RUN apt-get update && apt-get install -y \
+    libmagic1 \
+    libpq5 \
+    libsodium23 \
+    sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+
+FROM docker.io/python:${PYTHON_VERSION} AS build
+
+# install Poetry
+ARG POETRY_VERSION="1.3.2"
+RUN pip install poetry==${POETRY_VERSION}
+ARG POETRY_DYNAMIC_VERSIONING_VERSION="0.21.3"
+RUN poetry self add "poetry-dynamic-versioning[plugin]==${POETRY_DYNAMIC_VERSIONING_VERSION}"
+RUN poetry config virtualenvs.in-project true
+
+# add source files for building
+WORKDIR /src
+COPY pyproject.toml poetry.lock ./
+COPY scripts ./scripts/
+COPY tsdfileapi ./tsdfileapi/
+COPY .git ./.git/
+
+# create venv
+RUN python -m venv /app --upgrade-deps
+RUN /app/bin/pip install wheel
+
+# install dependencies
+RUN poetry export -f requirements.txt -o requirements.txt
+RUN /app/bin/pip install -r requirements.txt
+
+# build and install the file API
+RUN poetry build --no-interaction
+RUN /app/bin/pip install dist/*.whl
+
+
+FROM base AS run
+
+COPY --from=build /app /app
+
+EXPOSE 3003
+
+ENTRYPOINT [ "/app/bin/tsd-file-api" ]
```

Comment:
would be cool if one could make it possible to pass a config file instead of running with defaults, but not necessary in this PR

Reply:
This should be possible already, although I'll admit I've forgotten to test it. Since the file API is defined as an `ENTRYPOINT` rather than `CMD`, any commands specified when running the container will be passed to this entrypoint rather than a shell (or the Python interpreter, which is default for these base images).

I think this would work:

```console
docker run -p 3003:3003 -v /path/to/myconfig.yaml:/config.yaml:ro ghcr.io/unioslo/tsd-file-api /config.yaml
```

We could consider allowing the config path to be set via an env var too if it's not supported already.

Reply:
Ah nice, can you add this to the README?

Reply:
Will do. :+1: